electrumx/lib/tx.py

```diff
@@ -306,6 +306,93 @@ def read_tx_and_vsize(self):
         return tx, vsize
 
 
+class LitecoinMWTxError(Exception):
+    pass
+
+
+class DeserializerLitecoin(DeserializerSegWit):
+
+    def _read_tx_parts(self):
+        '''Return a (deserialized TX, tx_hash, vsize) tuple.'''
+        start = self.cursor
+        marker = self.binary[self.cursor + 4]
+        if marker:
+            # We could call super().read_tx here but the call stack is
+            # expensive when executed millions of times.
+            tx = Tx(
+                self._read_le_int32(),  # version
+                self._read_inputs(),    # inputs
+                self._read_outputs(),   # outputs
+                self._read_le_uint32()  # locktime
+            )
+            tx_hash = self.TX_HASH_FN(self.binary[start:self.cursor])
+            return tx, tx_hash, self.binary_length
+
+        version = self._read_le_int32()
+        orig_ser = self.binary[start:self.cursor]
+
+        marker = self._read_byte()
+        flag = self._read_byte()
+
+        # Work around a quirk of litecoind v0.21.2 and -rpcserialversion=1 that
+        # returns invalid stripped MW-only transactions with regular no inputs
+        # or outputs, causing this to look like a segwit transaction, but it
+        # really has zeros for the number of inputs and outputs, which we parsed
+        # as marker and flag. The next bytes are actually from lockTime. Ignore.
```

Comment:
if it is serialized as a segwit tx, then the number of inputs would follow the marker and flag, or not?

Reply:
What this is saying is that it's not serialized as a segwit transaction, but the logic that looks at what would be the segwit marker does not work for these invalid transactions.  At the top we do `if marker`, if it takes the non-segwit branch that's because it's inferring that there really is no segwit marker and what it read was actually the number of inputs.  In the case of this invalid txn, the number of inputs is _also zero_ so the standard logic of assuming we found the segwit marker if we see a zero fails.  Since [this is a bug that will be fixed in litecoind](https://github.com/spesmilo/electrumx/issues/179#issuecomment-1278272675), I think generally correct logic of detecting the segwit marker can stay.